README.md

```diff
@@ -335,10 +361,27 @@ Alternatively, extending or integrating backends (subclasses of [`curvature.curv
 approximations to the Laplace approximations.
 For example, currently the [`curvature.CurvlinopsInterface`](https://github.com/AlexImmer/Laplace/blob/main/laplace/curvature/curvlinops.py) based on [Curvlinops](https://github.com/f-dangel/curvlinops) and the native `torch.func` (previously known as `functorch`), [`curvature.BackPackInterface`](https://github.com/AlexImmer/Laplace/blob/main/laplace/curvature/backpack.py) based on [BackPACK](https://github.com/f-dangel/backpack/) and [`curvature.AsdlInterface`](https://github.com/AlexImmer/Laplace/blob/main/laplace/curvature/asdl.py) based on [ASDL](https://github.com/kazukiosawa/asdfghjkl) are available.
 
-The `curvature.CurvlinopsInterface` backend is the default and provides all Hessian approximation variants except the low-rank Hessian.
-For the latter, `curvature.AsdlInterface` can be used.
-Note that `curvature.AsdlInterface` and `curvature.BackPackInterface` are less complete and less compatible than `curvature.CurvlinopsInterface`.
-So, we recommend to stick with `curvature.CurvlinopsInterface` unless you have a specific need of ASDL or BackPACK.
+## When to use which backend
+
+> [!TIP]
+> Each backend as its own caveat/behavior. The use the following to guide you
+> picking the suitable backend, depending on you model & application.
+
+- **Small, simple MLP, or last-layer Laplace:** Any backend should work well.
+  `CurvlinopsGGN` or `CurvlinopsEF` is recommended if
+  `hessian_factorization = 'kron'`, but it's inefficient for other factorizations.
+- **LLMs with PEFT (e.g. LoRA):** `AsdlGGN` and `AsdlEF` are recommended.
+- **Continuous Bayesian optimization:** `CurvlinopsGGN/EF` and `BackpackGGN/EF` are
+  recommended since they are the only ones supporting backprop over Jacobians.
+
+> [!CAUTION]
+> The `curvlinops` backends are inefficient for full and diagonal factorizations.
+> Moreover, they're also inefficient for computing the Jacobians of large models
+> since they rely on `torch.func.jacrev` along `torch.func.vmap`!
+
+> [!CAUTION]
+> The `BackPack` backends are limited to models expressed as `nn.Sequential`.
+> Also, they're not compatible with normalization layers.
```

Comment:
curvlinops only computes K-FAC (`hessian_factorization = 'kron'`) for `nn.Linear` and `nn.Conv2d` modules.